trio/_wait_for_single_object.py

```diff
@@ -0,0 +1,56 @@
+from . import _timeouts
+from . import _core
+from ._threads import run_sync_in_worker_thread
+from ._core._windows_cffi import ffi, kernel32, ErrorCodes
+
+
+class StubLimiter:
+    def release_on_behalf_of(self, x):
+        pass
+
+    async def acquire_on_behalf_of(self, x):
+        pass
+
+
+async def WaitForSingleObject(handle):
```

Comment:
Okay, now that this is going to be a public API, we should think about what kind of object people will pass in as this `handle` object. Internally we use our ffi object's HANDLE objects, but the ffi object isn't part of our public API, so we can't expect users to do that. We should also let users pass in plain old integers, and then cast them to our HANDLE objects internally to use them.

The `_io_windows.py` has a `_handle` helper function that does this. We should move that helper into `_windows_cffi.py` (since it's part of our general "accessing the Windows API" code, not part of `WindowsIOManager` specifically), and then in this function add something like `handle = _handle(handle)`.